src/platform/linux/inputtino.cpp

```diff
@@ -0,0 +1,648 @@
+          break;
+        }
+        case LI_TOUCH_EVENT_CANCEL:
+        case LI_TOUCH_EVENT_UP:
+        case LI_TOUCH_EVENT_HOVER_LEAVE: {
+          (*raw->touch).release_finger(touch.pointerId);
+          break;
+        }
+          // TODO: LI_TOUCH_EVENT_CANCEL_ALL
+      }
+    }
+  }
+
+  static inline float
+  deg2rad(float degree) {
+    return degree * (M_PI / 180.f);
+  }
+
+  void
+  pen(client_input_t *input, const touch_port_t &touch_port, const pen_input_t &pen) {
+    auto raw = (client_input_raw_t *) input;
+    if (raw->pen) {
+      // First set the buttons
+      (*raw->pen).set_btn(inputtino::PenTablet::PRIMARY, pen.penButtons & LI_PEN_BUTTON_PRIMARY);
+      (*raw->pen).set_btn(inputtino::PenTablet::SECONDARY, pen.penButtons & LI_PEN_BUTTON_SECONDARY);
+      (*raw->pen).set_btn(inputtino::PenTablet::TERTIARY, pen.penButtons & LI_PEN_BUTTON_TERTIARY);
+
+      // Set the tool
+      inputtino::PenTablet::TOOL_TYPE tool;
+      switch (pen.toolType) {
+        case LI_TOOL_TYPE_PEN:
+          tool = inputtino::PenTablet::PEN;
+          break;
+        case LI_TOOL_TYPE_ERASER:
+          tool = inputtino::PenTablet::ERASER;
+          break;
+        default:
+          tool = inputtino::PenTablet::SAME_AS_BEFORE;
+          break;
+      }
+
+      // Normalize rotation value to 0-359 degree range
+      auto rotation = pen.rotation;
+      if (rotation != LI_ROT_UNKNOWN) {
+        rotation %= 360;
+      }
+
+      // Here we receive:
+      //  - Rotation: degrees from vertical in Y dimension (parallel to screen, 0..360)
+      //  - Tilt: degrees from vertical in Z dimension (perpendicular to screen, 0..90)
+      float tilt_x = 0;
+      float tilt_y = 0;
+      // Convert polar coordinates into Y tilt angles
+      if (pen.tilt != LI_TILT_UNKNOWN && rotation != LI_ROT_UNKNOWN) {
+        auto rotation_rads = deg2rad(rotation);
+        auto tilt_rads = deg2rad(pen.tilt);
+        auto r = std::sin(tilt_rads);
+        auto z = std::cos(tilt_rads);
+
+        tilt_x = std::atan2(std::sin(-rotation_rads) * r, z) * 180.f / M_PI;
+        tilt_y = std::atan2(std::cos(-rotation_rads) * r, z) * 180.f / M_PI;
+      }
+
+      (*raw->pen).place_tool(tool,
+        pen.x,
+        pen.y,
+        pen.eventType == LI_TOUCH_EVENT_DOWN ? pen.pressureOrDistance : -1,
+        pen.eventType == LI_TOUCH_EVENT_HOVER ? pen.pressureOrDistance : -1,
+        tilt_x,
+        tilt_y);
+    }
+  }
+
+  enum ControllerType {
+    XboxOneWired,
+    DualSenseWired,
+    SwitchProWired
+  };
+
+  int
+  alloc_gamepad(input_t &input, const gamepad_id_t &id, const gamepad_arrival_t &metadata, feedback_queue_t feedback_queue) {
+    ControllerType selectedGamepadType;
+
+    if (config::input.gamepad == "x360"sv) {
+      BOOST_LOG(info) << "Gamepad " << id.globalIndex << " will be Xbox 360 controller (manual selection)"sv;
+      selectedGamepadType = XboxOneWired;
```

Comment:
Should we add an option for xbox-one instead of using x360?

Reply:
I wasn't sure how this was handled in the UI so I've re-used what was already defined there. Ideally we should also add the Nintendo joypad which is now present with this PR. 
I should look further into this..

Reply:
I think my comment [here](https://github.com/LizardByte/Sunshine/pull/2261#discussion_r1530351369) should be enough to help with the UI.

I also notice you have `ds4` and `ps4` under `supported_gamepads` (and we have that in our Windows input file as well)... I'm not sure what the difference is, but `ps4` doesn't appear to be exposed in the UI.